dill/_dill.py

```diff
@@ -1700,6 +1700,27 @@ def _get_typedict_type(cls, clsdict, attrs, postproc_list):
         # clsdict.pop('__prepare__', None)
     return clsdict, attrs
 
+def _get_typedict_abc(obj, _dict, attrs, postproc_list):
+    if hasattr(abc, '_get_dump'):
+        (registry, _, _, _) = abc._get_dump(obj)
+        register = obj.register
+        postproc_list.extend((register, (reg(),)) for reg in registry)
+    elif hasattr(obj, '_abc_registry'):
+        registry = obj._abc_registry
+        register = obj.register
+        postproc_list.extend((register, (reg,)) for reg in registry)
+    else:
+        raise PicklingError("Cannot find registry of ABC %s", obj)
+
+    if '_abc_registry' in _dict:
+        del _dict['_abc_registry']
+        del _dict['_abc_cache']
+        del _dict['_abc_negative_cache']
+        # del _dict['_abc_negative_cache_version']
+    else:
+        del _dict['_abc_impl']
```

Comment:
I tend to use `pop` instead of `del` to delete, just because it's protected against the keyword not being there.

Reply:
Actually, strangely enough, [`del` is older than `pop`](https://docs.python.org/release/1.4/lib/node13.html#SECTION00316000000000000000). All Python implementations, including MicroPython, will have both, so this is not a concern.

I chose to use `del` because I already use the value of `_abc_impl` earlier in the function, so I know it exists. I guess switching to `pop` might be better for the other statements though.

Reply:
I decided to leave it as is. This is what I use and it makes sense based on the return values of the different functions/operators:

- `del a[·]`: when I know the key exists in the dictionary and don't need the value
- `a.pop(·)`: when I know the key exists in the dictionary but do need the value
- `a.pop(·, None)`: when I do not know if the key exists in the dictionary

Reply:
Sorry, I didn't mean that for you to think I meant I was worried that `del`  doesn't exist, I meant it in this sense:
``a.pop(·, None): when I do not know if the key exists in the dictionary``
The bug encountered with `del` is almost always that the key doesn't exist, so if the key might not (in some rare case) exist, then `pop` will protect you against that.